packages/node/test/parsers.test.ts

```diff
@@ -1,43 +1,45 @@
+import { StackFrame } from '@sentry/types';
 import * as fs from 'fs';
 
+import { ContextLines, resetFileContentCache } from '../src/integrations/contextlines';
 import * as Parsers from '../src/parsers';
 import * as stacktrace from '../src/stacktrace';
 import { getError } from './helper/error';
 
 describe('parsers.ts', () => {
   let frames: stacktrace.StackFrame[];
   let spy: jest.SpyInstance;
+  let contextLines: ContextLines;
+
+  async function addContext(frames: StackFrame[]): Promise<void> {
+    await contextLines.addToEvent({ exception: { values: [{ stacktrace: { frames } }] } });
+  }
 
   beforeEach(() => {
     spy = jest.spyOn(fs, 'readFile');
     frames = stacktrace.parse(new Error('test'));
-    Parsers.resetFileContentCache();
+    contextLines = new ContextLines();
+    resetFileContentCache();
   });
 
   afterEach(() => {
     jest.restoreAllMocks();
   });
 
   describe('lru file cache', () => {
-    test('parseStack with same file', done => {
+    test('parseStack with same file', async () => {
       expect.assertions(1);
+
       let mockCalls = 0;
-      void Parsers.parseStack(frames)
-        .then(_ => {
-          mockCalls = spy.mock.calls.length;
-          void Parsers.parseStack(frames)
-            .then(_1 => {
-              // Calls to readFile shouldn't increase if there isn't a new error
-              expect(spy).toHaveBeenCalledTimes(mockCalls);
-              done();
-            })
-            .then(null, () => {
-              // no-empty
-            });
-        })
-        .then(null, () => {
-          // no-empty
-        });
+      let parsedFrames = Parsers.parseStack(frames);
+      await addContext(parsedFrames);
+
+      mockCalls = spy.mock.calls.length;
+      parsedFrames = Parsers.parseStack(frames);
+      await addContext(parsedFrames);
+
+      // Calls to readFile shouldn't increase if there isn't a new error
+      expect(spy).toHaveBeenCalledTimes(mockCalls);
```

Comment:
Okay, annoyingly it won't let me use the suggestion feature here, so doing it manually.

Suggestion:
```js
      let parsedFrames = Parsers.parseStack(frames);
      await addContext(parsedFrames);

      const numCalls = spy.mock.calls.length;
      parsedFrames = Parsers.parseStack(frames);
      await addContext(parsedFrames);

     // Calls to `readFile` shouldn't increase if there isn't a new error to parse whose stacktrace contains a file we haven't yet seen
      expect(spy).toHaveBeenCalledTimes(numCalls);
```

Since it won't highlight them for you, three changes there:
- `s/mockCalls/numCalls` so it's clear it's a count and not an array of `MockCall` objects
- `numCalls` is now a const and is no longer initialized to `0`, since it doesn't need to be accessible from within two different`.then()` functions anymore
- I clarified the comment

(These same suggestions apply to the other tests, or course, too.)